### ISSUE TITLE
Programs using fenv do not compile with newlib

### ISSUE THREAD

**Issue by mneilly** [riscv/riscv-newlib#14]
I'm trying to compile a simple program for RISCV with newlib that uses fenv to set the rounding mode.

```
#include <cstdio>
#include <cmath>
#include <cfenv>

int main(int argc, char **argv)
{
    float fs1 = 2.0;
    float fs2 = 3.0;
    float f;

    std::fesetround(FE_TONEAREST);
    f = fs1 / fs2;
    printf("%f\n", f);

    std::fesetround(FE_TOWARDZERO);
    f = fs1 / fs2;
    printf("%f\n", f);

    std::fesetround(FE_UPWARD);
    f = fs1 / fs2;
    printf("%f\n", f);

    std::fesetround(FE_DOWNWARD);
    f = fs1 / fs2;
    printf("%f\n", f);

    return 0;
}
```

This compiles with the linux/glibc build of the toolchain: 

```
$ riscv64-unknown-linux-gnu-g++ -std=c++11 -o /tmp/log2 /tmp/log2.cc
$
```

but not with newlib:

```
$ riscv64-unknown-elf-g++ -std=c++11 -o /tmp/fenv-newlib /tmp/fenv-newlib.cc
/tmp/fenv-newlib.cc: In function 'int main(int, char**)':
/tmp/fenv-newlib.cc:11:10: error: 'fesetround' is not a member of 'std'
     std::fesetround(FE_TONEAREST);
          ^~~~~~~~~~
/tmp/fenv-newlib.cc:11:21: error: 'FE_TONEAREST' was not declared in this scope
     std::fesetround(FE_TONEAREST);
                     ^~~~~~~~~~~~
/tmp/fenv-newlib.cc:15:10: error: 'fesetround' is not a member of 'std'
     std::fesetround(FE_TOWARDZERO);
          ^~~~~~~~~~
/tmp/fenv-newlib.cc:15:21: error: 'FE_TOWARDZERO' was not declared in this scope
     std::fesetround(FE_TOWARDZERO);
                     ^~~~~~~~~~~~~
/tmp/fenv-newlib.cc:19:10: error: 'fesetround' is not a member of 'std'
     std::fesetround(FE_UPWARD);
          ^~~~~~~~~~
/tmp/fenv-newlib.cc:19:21: error: 'FE_UPWARD' was not declared in this scope
     std::fesetround(FE_UPWARD);
                     ^~~~~~~~~
/tmp/fenv-newlib.cc:23:10: error: 'fesetround' is not a member of 'std'
     std::fesetround(FE_DOWNWARD);
          ^~~~~~~~~~
/tmp/fenv-newlib.cc:23:21: error: 'FE_DOWNWARD' was not declared in this scope
     std::fesetround(FE_DOWNWARD);
                     ^~~~~~~~~~~
$
```

Looking at the preprocessor output reveals that for the newlib toolchain there is no output from processing cfenv:

```
...
# 3 "/tmp/fenv-newlib.cc" 2
# 1 "/opt/esperanto/minion-sim-toolchains/gcc-newlib/riscv64-unknown-elf/include/c++/7.0.1/cfenv" 1 3
# 32 "/opt/esperanto/minion-sim-toolchains/gcc-newlib/riscv64-unknown-elf/include/c++/7.0.1/cfenv" 3

# 33 "/opt/esperanto/minion-sim-toolchains/gcc-newlib/riscv64-unknown-elf/include/c++/7.0.1/cfenv" 3
# 4 "/tmp/fenv-newlib.cc" 2


# 5 "/tmp/fenv-newlib.cc"
int main(int argc, char **argv)
{
    float fs1 = 2.0;
    float fs2 = 3.0;
    float f;

    std::fesetround(FE_TONEAREST);
    f = fs1 / fs2;
    printf("%f\n", f);

    std::fesetround(FE_TOWARDZERO);
    f = fs1 / fs2;
    printf("%f\n", f);

    std::fesetround(FE_UPWARD);
    f = fs1 / fs2;
    printf("%f\n", f);

    std::fesetround(FE_DOWNWARD);
    f = fs1 / fs2;
    printf("%f\n", f);

    return 0;
}
```

Looking at the header file, it appears to be expecting the _GLIBCXX_USE_C99_FENV_TR1 macro to be defined:

```
$ less .../riscv-tools-newlib-stock/riscv64-unknown-elf/include/c++/7.1.1/cfenv
...
#if _GLIBCXX_USE_C99_FENV_TR1
...
namespace std
{
  // types
  using ::fenv_t;
  using ::fexcept_t;

  // functions
  using ::feclearexcept;
  using ::fegetexceptflag;
  using ::feraiseexcept;
  using ::fesetexceptflag;
  using ::fetestexcept;

  using ::fegetround;
  using ::fesetround;

  using ::fegetenv;
  using ::feholdexcept;
  using ::fesetenv;
  using ::feupdateenv;
} // namespace std

#endif // _GLIBCXX_USE_C99_FENV_TR1
...
```

**Comment by palmer-dabbelt** [comment]
It looks like most newlib targets don't provide a working fenv.h.  I'd be happy to accept a patch that adds one for RISC-V, but it's not a high priority thing right now.

**Comment by kito-cheng** [comment]
Implemented in #16 .
